File: app/src/main/java/com/abs/launcher/model/AppInfo.java
package com.abs.launcher.model;

import android.content.Intent;
import android.graphics.drawable.Drawable;

import com.abs.launcher.ItemInfo;

/**
 * Created by ZY on 2017/4/2.
 */

public class AppInfo {
    public long id = ItemInfo.NO_ID;
    public Intent intent;
    public boolean system;
    public boolean isShortcut;
    public int category;
    public int storage;

    public Drawable icon;
    public String title;

    public long lastUpdateTime;
    public long lastCalledTime;
    public int calledNums;


}

File: app/src/main/java/com/abs/launcher/model/AppInfo.kt
package com.abs.launcher.model

import android.content.Intent
import android.graphics.drawable.Drawable

/**
 * Created by zy on 17-12-17.
 */
data class AppInfo(var title: String, var icon: Drawable?, val intent: Intent) {
    var id = NO_ID
    var isSystem: Boolean = false
    var storage: Int = 0
    var category: Int = -1

    var iconResource: IconResource? = null

    val isShortcut = intent.component == null

    var usage: Usage = Usage()
}